repository/src/main/java/org/apache/atlas/repository/store/graph/v2/preprocessor/datamesh/DataProductPreProcessor.java

```diff
@@ -442,6 +442,9 @@ public void processDelete(AtlasVertex vertex) throws AtlasBaseException {
                     }
                 }
             }
+            if(RequestContext.get().getDeleteType() == DeleteType.SOFT){
```

Comment:
Also check if DeleteType.DEFAULT is set as SOFT also, right now default is soft delete so when we don't pass the parameter it will do soft delete by default .You will get `RequestContext.get().getDeleteType()` as `DEFAULT` so check for this condition also.

Reply:
TLDR; by default we soft delete the asset so when `RequestContext.get().getDeleteType() == DeleteType.DEFAULT`, handle the case.